src/EditorFeatures/CSharpTest/AddBraces/AddBracesTests.cs

```diff
@@ -1322,6 +1323,173 @@ static void Main()
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, false)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        [WorkItem(32480, "https://github.com/dotnet/roslyn/issues/32480")]
+        public async Task FireForIfWithoutIntercedingDirective(int bracesPreference, bool expectDiagnostic)
+        {
+            await TestAsync(
+   @"
+#define test
+class Program
+{
+    static void Main()
+    {
+#if test
+#endif
+        [|if (true)|]
+            return;
+    }
+}",
+
+   @"
+#define test
+class Program
+{
+    static void Main()
+    {
+#if test
+#endif
+        if (true)
+        {
+            return;
+        }
+    }
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, false)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        [WorkItem(32480, "https://github.com/dotnet/roslyn/issues/32480")]
+        public async Task FireForInnerNestedStatementWhenDirectiveEntirelyInside(int bracesPreference, bool expectDiagnostic)
+        {
+            await TestAsync(
+   @"
+#define test
+class Program
+{
+    static void Main()
+    {
+        [|while (true)
+#if test
+            if (true)|]
+                return;
+#endif
+    }
+}",
+
+   @"
+#define test
+class Program
+{
+    static void Main()
+    {
+        while (true)
+#if test
+            if (true)
+            {
+                return;
+            }
+#endif
+    }
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, true)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        [WorkItem(32480, "https://github.com/dotnet/roslyn/issues/32480")]
+        public async Task FireForOuterNestedStatementWhenDirectiveEntirelyInside(int bracesPreference, bool expectDiagnostic)
+        {
+            await TestAsync(
+   @"
+#define test
+class Program
+{
+    static void Main()
+    {
+        [|while (true)
+#if test
+            if (true)|]
+#endif
+                return;
+    }
+}",
+
+   @"
+#define test
+class Program
+{
+    static void Main()
+    {
+        while (true)
+        {
+#if test
+            if (true)
+#endif
+                return;
+        }
+    }
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
```

Comment:
need test with the `#if` after the `return`.  

What about some `if ... else` tests (without blocks) with interesting pp directives?

Reply:
Good call on the else case - there is actually some interesting behavior I didn't know about.

For example the following
```csharp
static void Main()
{
    if (true)
#if test
        return;
    else
        return;
#endif
}
```
In this case, we would expect parentheses around the else return and before the endif, like so. 
```csharp
static void Main()
{
    if (true)
#if test
        return;
    else
    {
        return;
    }
#endif
}
```
This works as expected.

What doesn't work is when the braces should be added to the if statement rather than the else, for example.
```csharp
static void Main()
{
#if test
    if (true)
        return;
    else
#endif
        return;
}
```
I'd expect to get this back
```csharp
static void Main()
{
#if test
    if (true)
    {
        return;
    }
    else
#endif
        return;
}
```

But actually no diagnostic is suggested to add braces to the if statement.

This behavior is caused because the else statement is a child of the if statement.  
When the directive intercedes the else, `ContainsIntercedingDirective` returns true even for the if statement (since else is a child and there is a directive inside) and true for else (expected).
When the directive intercedes the if, `ContainsIntercedingDirective` returns true for if (expected) and false for else (expected).

Fixing this means I'll probably have to at least partially roll my own function for `ContainsIntercedingDirective` to ignore the else if we're in an if statement.  I couldn't find anything obvious that already exists and would check for interceding directives between two specific nodes. 

@CyrusNajmabadi any thoughts on the above?

Reply:
> @CyrusNajmabadi any thoughts on the above?

That behavior seems acceptable to me.  I would simply track it with a test so that if we change things here in the future we can understand any implications of the change.  Thanks!

Reply:
I've been playing around with this for a while and may have found a relatively straightforward solution to solving that example - there's a private couple helpers around `ContainsInterleavedDirective` that would allow me to specify a span to check for interleaved directives rather than just passing in the entire node.  

Passing in only the IF and embedded statement as the span seems to be working so I'll add a few more tests and post it for review in a bit.  If it ends up not working I'll just use the version from this commit.

Reply:
Sounds good!